### PR TITLE
Fix #3432 Recurring expenses spawn at UTC timing instead of local time

### LINKED ISSUE #3432: Recurring expenses spawn at UTC timing instead of local time
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

The recurring expenses only spawn at 12AM UTC instead of respecting local timezone's 12AM. This mainly affects timezonw in which ahead of UTC +XX.

### To Reproduce

1. Create a recurring expenses
2. Wait for the local time hit 12am, but the expenses hasn't spawn.

### Expected behavior

The recurring expenses engine should respect local timezone and spawn at 12am daily (based on local timezone)

### Screenshots

_No response_

### App version

2024.08.25 (195) Play Store Version

### Smartphone

S24Ultra

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mushfiq1060**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## Screen recording of your changes (if applicable):
No

## What's changed?
Assume that on September 29, 2024, we create a recurring expense named **`MyRecurrentExpense`**. Consequently, an entry for **`MyRecurrentExpense`** appears in the **`upcoming section`** of the home screen for September 29. Previously, this entry was moved to the **`overdue section`** at 12:00 AM UTC rather than 12:00 AM local time. Now, the entry is moved to the **`overdue section`** at 12:00 AM local time.

**N.B:** Expenses from September 29 will move to the **`overdue section`** on September 30 at 12:00 AM local time.

## Risk factors

**What may go wrong if we merge your PR?**
No

**In what cases won't your code work?**
No

## Does this PR close any GitHub issues? (do not delete)

- Closes N/A

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/RecurringRuleModal.kt [hunk 1]
```diff
@@ -81,7 +81,7 @@ fun BoxWithConstraintsScope.RecurringRuleModal(
     onRuleChanged: (LocalDateTime, oneTime: Boolean, Int?, IntervalType?) -> Unit,
 ) {
     var startDate by remember(modal) {
-        mutableStateOf(modal?.initialStartDate ?: timeNowUTC())
+        mutableStateOf(modal?.initialStartDate ?: timeNowLocal())
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/RecurringRuleModal.kt:84]: Use `LocalTimeProvider` composition local to get the current time instead

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/action/viewmodel/home/DueTrnsInfoAct.kt [hunk 2]
```diff
@@ -16,20 +16,24 @@ import com.ivy.wallet.domain.pure.exchange.exchangeInBaseCurrency
...
+    private val timeConverter: TimeConverter
 ) : FPAction<DueTrnsInfoAct.Input, DueTrnsInfoAct.Output>() {
 
     override suspend fun Input.compose(): suspend () -> Output =
         suspend {
             range
         } then dueTrnsAct then { trns ->
-            val dateNow = dateNowUTC()
+            val dateNow = with(timeConverter) {
+                Instant.now().toLocalDate()
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/domain/action/viewmodel/home/DueTrnsInfoAct.kt:35]: Use `TimeProvider`

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]

**Comment by ILIYANGERMANOV** [comment]
@Mushfiq1060 add screen recording to prove that it worms

**Comment by Mushfiq1060** [comment]
ok

**Comment by Mushfiq1060** [comment]
@ILIYANGERMANOV screen recording

https://github.com/user-attachments/assets/dc92f79d-3ec5-4b9b-8793-525c15ebbcc5

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/RecurringRuleModal.kt [hunk 3]
```diff
@@ -80,8 +80,9 @@ fun BoxWithConstraintsScope.RecurringRuleModal(
     dismiss: () -> Unit,
     onRuleChanged: (LocalDateTime, oneTime: Boolean, Int?, IntervalType?) -> Unit,
 ) {
+    val localTimeProvider = LocalTimeProvider.current
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/RecurringRuleModal.kt:83]: just call it timeProvider

**Review by ILIYANGERMANOV — APPROVED** [review]

**Review by shamim-emon — APPROVED** [review]

**Pull request merged**